### ISSUE TITLE
Confusion of rate and shape

### ISSUE THREAD

**Issue by ghost** [dkahle/invgamma#1]
I may be wrong here, but according to the [Wikipedia page](https://en.wikipedia.org/wiki/Inverse-gamma_distribution), if X is gamma-distributed with shape _a_ and rate _b_, then 1/X is IG-distributed with shape _a_ and rate 1/_b_ (and thus scale _b_). The converse holds true as well. But it seems to me that in your function dinvgamma, for example, the rate of the IG distribution is used directly as the rate in the function stats::dgamma, without first taking the reciprocal.

**Comment by Achab94** [comment]
@BenjaK issue is extremely important and @dkahle you should take into consideration on changing the misleading _rate_ and _shape_ terms in the documentation of the package (as in the arguments of the function).

**Comment by jarad** [comment]
Also, a [scale parameter](https://en.wikipedia.org/wiki/Scale_parameter) has a technical definition which is why the [wikipedia page for inverse gamma](https://en.wikipedia.org/wiki/Inverse-gamma_distribution#Probability_density_function) uses the parameter "scale".

**Comment by tianyi21** [comment]
This is true! The definitions of rate and scale in `rinvgamma` are also kind of mixed up :(

**Comment by dkahle** [comment]
Hi all – thanks for the comments, sorry for taking so long to respond. And sorry for the confusion!

The point of **invgamma** was and is to provide a simple, more or less self-maintaining implementation of the `d`, `p`, `q`, and `r` functions of the inverse gamma distribution.  To do this, it uses [the transformation theorem](https://en.wikipedia.org/wiki/Random_variable#Functions_of_random_variables) along with the implementations of the corresponding functions for the gamma distribution, which are already implemented in **stats** as `dgamma()` and so on. That's why if you look at the bodies of the functions, e.g. `dinvgamma()`, they are all wrappers of their corresponding gamma counterparts, e.g. `dgamma()`. The argument names were intended to reflect this fact. Thus, `rate` in `dinvgamma()` refers to the `rate` parameter of `dgamma()`.

The problem presented here is this: the parameter `rate` of `dgamma()` is called `rate` because it is a rate parameter, which [has a technical definition](https://en.wikipedia.org/wiki/Location–scale_family) in a much broader context than the gamma/inverse gamma family. In fact, the inverse gamma distribution has a rate parameter in this formal sense as well, but that rate parameter is not the same as the rate parameter of the gamma distribution, as @BenjaK rightly points out. To be clear:  the argument `rate` in **invgamma**'s functions is *not* the rate parameter of the inverse gamma distribution. In fact, it is its inverse.

An analogy may be helpful here. The [lognormal distribution](https://en.wikipedia.org/wiki/Log-normal_distribution) $LogN(\mu, \sigma^2)$ is, by definition, the distribution of a random variable whose log is a normal $N(\mu, \sigma^2)$ distribution, i.e. $Y = e^{X}$ with $X \sim N(\mu, \sigma^2)$: a lognormal is the exponential of a normal. Throughout statistics $\mu$ conventionally denotes the mean of a RV/distribution and $\sigma^2$ its variance. However, the mean of a lognormal random variable is not $\mu$, and $\sigma^2$ is not its variance: you just have to know that they're the mean and variance of the "distribution behind the distribution." And that's a little confusing and goes against the convention that $\mu$ ( $\sigma^2$) is the mean (variance) of the distribution. It's a clash of conventions.

Base R implements the lognormal distribution as `dlnorm()` with formal arguments `meanlog` and `sdlog` to try to clarify this distinction, but I didn't do that when writing **invgamma**, and that's what's causing the confusion. When I was writing **invgamma** my intention was in fact to write other similar packages that do the same thing: provide functions for other common distributions not in R that are simple transformations of distributions that are. My convention was to maintain the parameters of the "distributions behind the distributions." I still think that makes sense, but it does make this part messy.

To avoid mishaps, I have added a package startup message to this effect. But I feel like it's not really something I can change at this point, because everyone who has been using it all of a sudden will have breaking code with no warning (the parameter spaces are the same, so the code itself can't determine the coder's intent). So, sorry for the confusion! I hope this note helps.

**Comment by dkahle** [comment]
See 0fc367b for details.
